_posts/2016-11-14-JS-Glossary.md

```diff
@@ -0,0 +1,554 @@
+
+Babel's plugins can be configured inside your `Package.json`. To ship your code to the world, you then create a script of some sort to convert your source code into "olde world" JavaScript via Babel.
+
+In the case of a react-native project, Babel is happening behind the scenes.
+
+### ESLint
+
+How can you be sure your syntax is correct? JavaScript has a really powerful and extensible linter called ESLint. If parses your JavaScript and offers warnings and errors around your syntax. You can use this to provide a consistent codebase, or in my case, to be lazy with your formatting. Fixing a lot of issues is one command away. 
+
+# Development
+
+### Live Reload
+
+This is a common feature in JavaScript tooling. If you press save in a source file then some action is taken. Live Reloading tends to be a more blunt action, for example reloading the current view from scratch, or running all of the tests related to the file.  
+
+### Hot-Reloading
+
+Hot Reloading is more rare, because it's significantly harder. Hot Reloading for React projects is injecting new functions into the running application, and keeping it in the same state. 
+
+For example if you had a filled-in form on your screen, you could make styling changes inside your source file and the text inside the form would not change. Hot reloading is amazing.
+
+### Haste Map
+
+Part of what makes React Native support Hot Reloading, and allows [Jest](#jest) to understand changes for testing is by using a Haste Map. A Haste Map is a dependency resolver for JavaScript, looking through every function to know how it connects to every other function within the JavaScript project. 
+
+With the dependencies mapped, it becomes possible to know what functions would need replacing or testing when you press save after writing some changes. This is why it takes a bit of time to start up a React Native project. 
+
+The public API is deprecated, you shouldn't use it in your projects, but the [old README is still around][haste].
+
+# Testing
+
+### Jest
+
+Facebook have their own test runner called Jest. It builds on [Jasmine][jasmine], and offers a few features that kick ass for me:
+
+* Re-runs failing tests first
+* Assumes all tests unrelated to changes are green and doesn't run them
+* Watch mode that works reliably 
+
+I miss these features when I'm not in a Jest project.
+
+### Jest Snapshots
+
+Jest also has a feature called Jest Snapshots, this allows you to take "snapshots" of JavaScript objects, and then verify they are they are the same as they were last time. In iOS we [used visual snapshot][snapshots] testing a lot.
+
```

Comment:
`has a feature called Jest Snapshots, this allows you to take "snapshots" of JavaScript objects` -> `has a feature called Jest Snapshots, that allows you to take "snapshots" of JavaScript objects`